File: LeetCode/LongestSubstring.cs
using System;
using System.Collections.Generic;
using System.Reflection;
using System.Text;

namespace LeetCode
{
    public class LongestSubstring
    {
        /// <summary>
        /// Sliding window solution
        /// </summary>
        /// <param name="s"></param>
        /// <returns></returns>
        public int LengthOfLongestSubstring(string s)
        {
            int res = 0;
            //hold unique chars
            HashSet<int> visited = new HashSet<int>();
            int l = 0; //left pointer
            int r = 0; //right pointer
            char[] chars = s.ToCharArray();
            //loop through chars, keep track of longest sub string
            while (r < chars.Length)
            {
                if (visited.Contains(chars[r]))
                {
                    visited.Remove(chars[l]);
                    l++;
                }
                else 
                {
                    visited.Add(chars[r]);
                    r++;
                    res = Math.Max(visited.Count, res);
                }

            }

            return res;
        }
    }
}

File: TypeScript/src/LongestSubstring.ts

export class LongestSubstring {

    public LengthOfLongestSubstring(s: string): number {
        let res: number = 0;
        let visited: Set<string> = new Set<string>();
        let l: number = 0;
        let r: number = 0;
        //let chars: Array<string> = new Array(s);
        while(r < s.length){
            if (visited.has(s[r])){
                visited.delete(s[l]);
                l++;
            } else {
                visited.add(s[r]);
                res = Math.max(Array.from(visited).length, res);
                r++;
            }
        }
        return res;
    }
}